pkg/controllers/cspi-controller/handler.go

```diff
@@ -413,3 +429,28 @@ func (c *CStorPoolInstanceController) addDiskUnavailableCondition(cspi *cstor.CS
 		}
 	}
 }
+
+// validateCSPI validates the CSPI and returns error
+func validateCSPI(cspi *cstor.CStorPoolInstance) error {
+	if len(cspi.Spec.DataRaidGroups) == 0 {
+		return errors.Errorf("No data RaidGroups exists")
+	}
+	if cspi.Spec.PoolConfig.DataRaidGroupType == "" {
+		return errors.Errorf("Missing DataRaidGroupType")
+	}
+	if len(cspi.Spec.WriteCacheRaidGroups) != 0 &&
+		cspi.Spec.PoolConfig.WriteCacheGroupType == "" {
+		return errors.Errorf("Missing WriteCacheRaidGroupType")
+	}
+	for _, rg := range cspi.Spec.DataRaidGroups {
+		if len(rg.CStorPoolInstanceBlockDevices) == 0 {
+			return errors.Errorf("No BlockDevices exist in DataRaidGroup")
```

Comment:
`"No BlockDevices exist in DataRaidGroup"`  ---> `"No BlockDevices exist in one of the DataRaidGroup"`

Reply:
Updated